code/modules/mob/living/living.dm

```diff
@@ -1130,21 +1130,41 @@
 		return TRUE
 	return FALSE
 
-/mob/living/proc/ExtinguishMob()
-	if(on_fire)
-		on_fire = FALSE
-		fire_stacks = 0
-		for(var/obj/effect/dummy/lighting_obj/moblight/fire/F in src)
-			qdel(F)
-		clear_alert("fire")
-		SEND_SIGNAL(src, COMSIG_CLEAR_MOOD_EVENT, "on_fire")
-		SEND_SIGNAL(src, COMSIG_LIVING_EXTINGUISHED, src)
-		update_fire()
+/**
+ * Extinguish all fire on the mob
+ *
+ * This removes all fire stacks, fire effects, alerts, and moods
+ * Signals the extinguishing.
+ */
+/mob/living/proc/extinguish_mob()
+	if(!on_fire)
+		return
+	on_fire = FALSE
+	fire_stacks = 0
+	for(var/obj/effect/dummy/lighting_obj/moblight/fire/F in src)
+		qdel(F)
+	clear_alert("fire")
+	SEND_SIGNAL(src, COMSIG_CLEAR_MOOD_EVENT, "on_fire")
+	SEND_SIGNAL(src, COMSIG_LIVING_EXTINGUISHED, src)
+	update_fire()
 
-/mob/living/proc/adjust_fire_stacks(add_fire_stacks) //Adjusting the amount of fire_stacks we have on person
-	fire_stacks = clamp(fire_stacks + add_fire_stacks, -20, 20)
-	if(on_fire && fire_stacks <= 0)
-		ExtinguishMob()
+/**
+ * Adjust the amount of fire stacks on a mob
+ *
+ * This modifies the firestacks on a mob, stacks are clamped between -20 and 20.
+ * If the fire stacks are reduced to 0 then we will extinguish the mob.
+ *
+ * Vars:
+ * * add_fire_stacks: int The amount to modify the fire stacks
+ * * set_stack: (optional)(default: false) Sets the fire stacks to the add_fire_stacks value instead of modifying it
+ */
+/mob/living/proc/adjust_fire_stacks(add_fire_stacks, set_stack=FALSE) //Adjusting the amount of fire_stacks we have on person
```

Comment:
Make a second proc, `set_fire_stacks()`, instead of having the `set_stack` parameters.
Look at the base of `/mob/proc/set_stat(new_stat)` and try to follow the format. No need to send a signal if you don't want it. For other examples, see the base of `setGrabState()`, `setMovetype()`, `set_anchored()`, `set_pulledby()` or various others.

Reply:
New proc added.